src/rpc/rawtransaction.cpp

```diff
@@ -43,14 +44,14 @@
 
 #include <univalue.h>
 
-static void TxToJSON(const CTransaction& tx, const uint256 hashBlock, UniValue& entry, CChainState& active_chainstate)
+static void TxToJSON(const CTransaction& tx, const uint256 hashBlock, UniValue& entry, CChainState& active_chainstate, const CTxUndo* txundo = nullptr)
 {
     // Call into TxToUniv() in bitcoin-common to decode the transaction hex.
     //
     // Blockchain contextual information (confirmations and blocktime) is not
     // available to code in bitcoin-common, so we query them here and push the
     // data into the returned UniValue.
-    TxToUniv(tx, uint256(), entry, true, RPCSerializationFlags());
+    TxToUniv(tx, uint256(), entry, true, RPCSerializationFlags(), txundo, TxVerbosity::SHOW_DETAILS_AND_PREVOUT);
```

Comment:
I understand that functionally this works fine, but it feels unnecessarily confusing to always pass `TxVerbosity::SHOW_DETAILS_AND_PREVOUT` even in the case when in actuality we're trying to get `TxVerbosity::SHOW_DETAILS` but relying on `txundo` being null to achieve that. Would it not be more prudent to instead add `verbosity` as an optional argument to `TxToJSON` and simply pass that along?

Reply:
I added an arg for explicit `TxVerbosity` in `TxToJSON`